### DISCUSSION TITLE
Metrics exporter collection interval

### DISCUSSION THREAD

**Discussion by sarkisjad** [open-telemetry/opentelemetry-java-instrumentation#12441] [Q&A] [answered]
Hello, 

I am currently using an extension to create a custom metrics exporter (https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/README.md).
I was wondering if there was a way to set the collection interval from the extension as opposed to put it as an environment variable as documented here 
https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#periodic-exporting-metricreader

I am asking because I can only know the collection interval I want to set from inside the extension.

Could you please let me know? 
Regards,

**Comment by trask** [comment, marked as answer]
yes! you can add your own properties supplier:

https://github.com/open-telemetry/opentelemetry-java/blob/e4d5109c496dce9fb2341711e4fbc0c888ecd785/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java#L94-L95

**Reply by sarkisjad** [reply]
Hello. Thanks for your reply. 
In my application, I’m using both manual instrumentation and auto instrumentation.

If I use this class, how can I make sure this property I'm setting is only used by the auto instrumentation? 
Is service loading (@AutoService) enough or should I do something else? 

Thanks

**Reply by trask** [reply]
> I am currently using an extension to create a custom metrics exporter

I think you just need to add it to your extension (along with the appropriate meta-inf/services file via `@AutoService` or similar)
